File: runme/mcgen/local/datan/datsrc/lsqp2g.c
#include <math.h>
#include "datsrc.h"

double LIBFUNC lsqp2g_(double const P_T *x, integer const P_T *nr,
               double const P_T *t)
{
    /* System generated locals */
    double ret_val, d__1, d__2;

    /* Local variables */
    double back, gauss1, gauss2, arg1, arg2;

    /* Parameter adjustments */
    --x;

    /* Function Body */
/* Computing 2nd power */
    d__1 = *t;
    back = x[1] + x[2] * *t + x[3] * (d__1 * d__1);
/* Computing 2nd power */
    d__1 = x[5] - *t;
/* Computing 2nd power */
    d__2 = x[6];
    arg1 = d__1 * d__1 / (d__2 * d__2 * 2.);
/* Computing 2nd power */
    d__1 = x[8] - *t;
/* Computing 2nd power */
    d__2 = x[9];
    arg2 = d__1 * d__1 / (d__2 * d__2 * 2.);
    if (arg1 > 700.) {
        gauss1 = 0.;
    } else {
        gauss1 = x[4] * exp(-arg1);
    }
    if (arg2 > 700.) {
        gauss2 = 0.;
    } else {
        gauss2 = x[7] * exp(-arg2);
    }
    ret_val = back + gauss1 + gauss2;
    return ret_val;
} /* lsqp2g_ */

File: runme/mcgen/local/datan/datsrc/lsqp2g.f
      DOUBLE PRECISION FUNCTION LSQP2G(X,NR,T)
      IMPLICIT DOUBLE PRECISION (A-H,O-Z)
      DIMENSION X(NR)
      PARAMETER (BIG=700.D0,TWO=2.D0,ZERO=0.D0)
      BACK=X(1)+X(2)*T+X(3)*T**2
      ARG1=(X(5)-T)**2/(TWO*X(6)**2)
      ARG2=(X(8)-T)**2/(TWO*X(9)**2)
      IF(ARG1.GT.BIG) THEN
        GAUSS1=ZERO
      ELSE
        GAUSS1=X(4)*EXP(-ARG1)
      END IF
      IF(ARG2.GT.BIG) THEN
        GAUSS2=ZERO
      ELSE
        GAUSS2=X(7)*EXP(-ARG2)
      END IF
      LSQP2G=BACK+GAUSS1+GAUSS2
      END
